src/LinAlg/hiopVectorRajaPar.cpp

```diff
@@ -855,6 +855,38 @@ void hiopVectorRajaPar::component_max(const hiopVector& vec)
   );
 }
 
+/**
+ * @brief Set each component to its absolute value
+ */
+void hiopVectorRajaPar::component_abs ()
+{
+  double* dd = data_dev_;
+  RAJA::forall< hiop_raja_exec >( RAJA::RangeSegment(0, n_local_),
+    RAJA_LAMBDA(RAJA::Index_type i)
+    {
+      dd[i] = fabs(dd[i]);
+    });
+}
+
+/**
+ * @brief Set each component to its absolute value
+ */
+void hiopVectorRajaPar::component_sgn ()
+{
+  double* dd = data_dev_;
+  RAJA::forall< hiop_raja_exec >( RAJA::RangeSegment(0, n_local_),
+    RAJA_LAMBDA(RAJA::Index_type i)
+    {
+      if(dd[i]<0.0) {
+        dd[i] = -1.0;
+      } else if(dd[i]>0.0) {
+        dd[i] = 1.0;
+      } else {
+        dd[i] = 0.0;
+      }
+    });
```

Comment:
Consider refactoring to have same instruction on each thread. Something like this may be a good solution:
```suggestion
    RAJA_LAMBDA(RAJA::Index_type i)
    {
      int sign = (0.0 < dd[i]) - (dd[i] < 0.0);
      dd[i] = static_cast<double>(sign);
    });
```

Reply:
Done!